### PR TITLE
xeus: init at 3.0.5

### PR DESCRIPTION
**Pull request by serge-sans-paille**
###### Description of changes

Initial upload + adding myself as a maintainer

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by NickCao — COMMENTED** [review]
Overall looks nice, but the indentation is wrong, better use an editor supporting editorconfig.

**Comment by serge-sans-paille** [comment]
@NickCao indentation plugin added, that should look better now (?)

File: pkgs/development/libraries/xeus/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "xeus";
+  version = "3.0.5";
+
+  src = fetchFromGitHub {
+    owner = "jupyter-xeus" ;
+    repo = pname ;
+    rev = version ;
+    sha256 = "sha256-LeU4PJ1UKfGqkRKq0/Mn9gjwNmXCy0/2SbjWJrjlOyU=" ;
```
Comment (NickCao) [inline pkgs/development/libraries/xeus/default.nix:19]: Something is still wrong with your editor (the empty space)?

File: pkgs/development/libraries/xeus/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    xtl
+  ];
+
+  cmakeFlags = [
+    "-DXEUS_BUILD_TESTS=ON"
+  ];
+  doCheck = true;
+  preCheck = ''
+        export LD_LIBRARY_PATH=$PWD
+  '';
```
Comment (NickCao) [inline pkgs/development/libraries/xeus/default.nix:39]: Try nixpkgs-fmt.

File: pkgs/development/libraries/xeus/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  buildInputs = [
+    nlohmann_json
+    libuuid
+    xtl
+  ];
+
+  cmakeFlags = [
+    "-DXEUS_BUILD_TESTS=ON"
+  ];
+  doCheck = true;
```
Comment (NickCao) [inline pkgs/development/libraries/xeus/default.nix:36]: Leave a empty line here as a logical separation between build and check.
```suggestion
  ];

  doCheck = true;
```

**Review by NickCao — APPROVED** [review]

**Pull request merged**